### PR TITLE
Replace most uses of QueryInfo with BasicQueryInfo

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/connector/system/KillQueryProcedure.java [hunk 1]
```diff
@@ -55,19 +55,20 @@ public void killQuery(String queryId, String message)
         QueryId query = parseQueryId(queryId);
 
         try {
-            QueryInfo queryInfo = queryManager.getQueryInfo(query);
+            Optional<QueryState> state = queryManager.getQueryState(query);
```
[the remark this answers is not in the document]
Comment (dain) [reply presto-main/src/main/java/com/facebook/presto/connector/system/KillQueryProcedure.java]: It is an existing method.  The query manager expires and removes queries over time automatically, so it is possible to ask for one that no longer exists.  Of course there real issue is that many methods throw, some ignore (idempotent methods like cancel), and two return optional if the query doesn't exist.  At the very least we should document the behavior in the face of a non-existant query.  As for changing the existing APIs, I think that is a bigger discussion.

**Comment by dain** [comment]
I applied all the comments, and added two commits at the end of the with the API changes you suggested and more documentation to SqlQueryManager.

File: presto-main/src/main/java/com/facebook/presto/connector/system/KillQueryProcedure.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (raghavsethi) [reply presto-main/src/main/java/com/facebook/presto/connector/system/KillQueryProcedure.java]: Fair enough.

File: presto-main/src/main/java/com/facebook/presto/connector/system/KillQueryProcedure.java [hunk 2]
```diff
@@ -55,19 +55,20 @@ public void killQuery(String queryId, String message)
...
                 throw new PrestoException(NOT_SUPPORTED, "Target query is not running: " + queryId);
             }
 
             queryManager.failQuery(query, createKillQueryException(message));
 
             // verify if the query was killed (if not, we lost the race)
-            queryInfo = queryManager.getQueryInfo(query);
-            if ((queryInfo.getState() != QueryState.FAILED) ||
-                    !ADMINISTRATIVELY_KILLED.toErrorCode().equals(queryInfo.getErrorCode())) {
+            if (!ADMINISTRATIVELY_KILLED.toErrorCode().equals(queryManager.getQueryInfo(query).getErrorCode())) {
```
Comment (raghavsethi) [inline presto-main/src/main/java/com/facebook/presto/connector/system/KillQueryProcedure.java]: This looks like maybe it's a behavior change (could queryManager.failQuery still be executing?), but I think that's fine.

File: presto-main/src/main/java/com/facebook/presto/connector/system/KillQueryProcedure.java [hunk 3]
```diff
@@ -26,7 +26,6 @@
 
 import java.lang.invoke.MethodHandle;
 import java.util.NoSuchElementException;
-import java.util.Optional;
```
Comment (raghavsethi) [inline presto-main/src/main/java/com/facebook/presto/connector/system/KillQueryProcedure.java]: Remove period from end of commit title

**Pull request closed without merging**